### PR TITLE
feat(input-reader):create input-reader

### PR DESCRIPTION
**Pull request by fmoura**

### CODE REVIEW

File: cmd/cartesi-rollups-cli/root/send/send.go [hunk 1]
```diff
@@ -29,6 +31,7 @@ var (
 	account         uint32
 	hexPayload      string
 	addressBookFile string
+	verbose         bool
```
Comment (marcelstanley) [inline cmd/cartesi-rollups-cli/root/send/send.go:34]: Perhaps we should move the cli changes to a separate commit.

File: internal/inputreader/inputreader.go [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+	ctx context.Context,
+	ready chan<- struct{},
+) error {
+	// Check the last block processed by the the Input Reader
+	previouslyKnownMostRecentFinalizedBlockNumber, err := r.repository.GetMostRecentFinalizedBlockNumber(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Safegaurd: Only check blocks after InputBox was deployed
```
Comment (marcelstanley) [inline internal/inputreader/inputreader.go:96]:
```suggestion
	// Safeguard: Only check blocks after InputBox was deployed
```

File: internal/inputreader/inputreader.go [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+		applicationAddress:  applicationAddress,
+	}
+}
+
+func (r InputReader) Start(
+	ctx context.Context,
+	ready chan<- struct{},
+) error {
+	// Check the last block processed by the the Input Reader
+	previouslyKnownMostRecentFinalizedBlockNumber, err := r.repository.GetMostRecentFinalizedBlockNumber(ctx)
```
Comment (marcelstanley) [inline internal/inputreader/inputreader.go:91]:
```suggestion
	previousMostRecentFinalizedBlockNumber, err := r.repository.GetMostRecentFinalizedBlockNumber(ctx)
```

File: pkg/ethutil/ethutil.go [hunk 4]
```diff
@@ -106,6 +108,12 @@ func getInputIndex(
 		if err != nil {
 			return 0, fmt.Errorf("failed to parse input added event: %v", err)
 		}
+
```
Comment (marcelstanley) [inline pkg/ethutil/ethutil.go:111]: Perhaps this change also should be moved to a separate commit

**Review by marcelstanley — COMMENTED** [review]
Made some suggestions after a very quick look:

**Review by GMKrieger — COMMENTED** [review]
Most of the code seems fine, although I couldn't run the tests on my machine

There are only 2 major comments:
- All of the json files are either missing the EOF linebreak or have 4+ lines at the end
- There should be multiple commits, as Marcel mentioned. IMHO, there should be a commit for the cli changes, one for the ethutil changes, one for the input reader and one for the tests

**Comment by fmoura** [comment]
Created PR #455 with cli commits

**Comment by fmoura** [comment]
This is being deprecated in favor of https://github.com/cartesi/rollups-node/pull/512
